File: 프로그래머스/0/181857. 배열의 길이를 2의 거듭제곱으로 만들기/배열의 길이를 2의 거듭제곱으로 만들기.js
function solution(arr) {
    let answer = [...arr];
    let minLength = 1;
    
    while (minLength < arr.length) {
        minLength *= 2;
    }
    
    for (let i = 0; i < (minLength - arr.length); i++) {
        answer.push(0);
    }
    return answer;
}

File: 프로그래머스/0/181857. 배열의 길이를 2의 거듭제곱으로 만들기/배열의 길이를 2의 거듭제곱으로 만들기.py
def solution(arr):
    answer = list(arr)
    min_length = 1

    while min_length < len(arr):
        min_length *= 2

    num_zeros_to_add = min_length - len(arr)
    answer.extend([0] * num_zeros_to_add)
    return answer